sdk/iothub/src/service/requests/get_identity_builder.rs

```diff
@@ -0,0 +1,61 @@
+use std::convert::{TryFrom, TryInto};
+use std::marker::PhantomData;
+
+use bytes::Bytes;
+use http::{Method, Response, StatusCode};
+
+use crate::service::{IoTHubError, ServiceClient, API_VERSION};
+
+/// The GetIdentityBuilder is used to construct a request for creating
+/// or updating a module or device identity.
+pub struct GetIdentityBuilder<'a, T>
+where
+    T: TryFrom<Response<Bytes>, Error = IoTHubError>,
+{
+    service_client: &'a ServiceClient,
+    device_id: String,
+    module_id: Option<String>,
+    return_type: PhantomData<T>,
+}
+
+impl<'a, T> GetIdentityBuilder<'a, T>
+where
+    T: TryFrom<Response<Bytes>, Error = IoTHubError>,
+{
+    pub(crate) fn new(
+        service_client: &'a ServiceClient,
+        device_id: String,
+        module_id: Option<String>,
+    ) -> Self {
+        Self {
+            service_client,
+            device_id,
+            module_id,
+            return_type: PhantomData,
+        }
+    }
+
+    /// Execute the request to create or update the module or device identity.
+    pub async fn execute(&self) -> Result<T, IoTHubError> {
```

Comment:
I'm not sure how I feel about these builders that aren't really builders (i.e., they just have an `execute` method and don't take optional configuration). This is certainly more uniform but it does make the user type an additional `execute()` invocation for no real purpose.

Reply:
I agree, I changed this back again to be able to invoked without `execute()`.